tests/rptest/tests/log_segment_ms_test.py

```diff
@@ -0,0 +1,196 @@
+import typing
+
+from ducktape.mark import parametrize, defaults
+from ducktape.utils.util import wait_until
+
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.verifiable_consumer import VerifiableConsumer
+from rptest.services.verifiable_producer import VerifiableProducer
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.util import segments_count
+
+# NOTE this value could be read from redpanda
+# for example this should be a reasonable way
+# int(Admin(self.redpanda).get_cluster_config_schema()
+#                       ['properties']['log_segment_ms_min']['example'])
+SERVER_SEGMENT_MS = 60000
+SERVER_HOUSEKEEPING_LOOP = 10
+
+
+def messages_to_generate_segments(num_segments, period):
+    return int(period / 1000 * (num_segments) + SERVER_HOUSEKEEPING_LOOP + 1)
+
+
+TEST_NUM_SEGMENTS = 3
+ERROR_MARGIN = 1
```

Comment:
Why is 1 the correct value for error margin?

Reply:
increased TEST_NUM_SEGMENTS to 5, ERROR_MARGIN set to 1 accounts for the fact that housekeeping loop has granularity of 10 seconds, so the segment could be ready for rolling but the housekeeping task did not run yet